examples/MatLabAgent_v2/mlaV2/agent.py

```diff
@@ -0,0 +1,178 @@
+    the agent created using that configuration.
+
+    :param config_path: Path to a configuration file.
+
+    :type config_path: str
+    :returns: Mlav2
+    :rtype: Mlav2
+    """
+    try:
+        config = utils.load_config(config_path)
+    except StandardError:
+        config = {}
+
+    if not config:
+        _log.info("Using Agent defaults for starting configuration.")
+
+    script_names = config.get('script_names', ["test.py"])
+    script_args = config.get('script_args', [["20"]])
+    topics_to_agents= config.get('topics_to_agents', ["matlab/to_agent/1"])
+    topics_from_agents = config.get('topics_from_agents', "matlab/from_agent/")
+
+
+    return Mlav2(script_names, script_args, topics_to_agents, topics_from_agents, **kwargs)
+
+
+class Mlav2(Agent):
+    """
+    Document agent constructor here.
+    """
+
+    def __init__(self,script_names=[], script_args=[], topics_to_agents=[], 
+            topics_from_agents=None,**kwargs):
+
+        super(Mlav2, self).__init__(**kwargs)
+        _log.debug("vip_identity: " + self.core.identity)
+
+        self.script_names = script_names
+        self.script_args = script_args
+        self.topics_to_agents = topics_to_agents
+        self.topics_from_agents = topics_from_agents
+        self.default_config = {"script_names": script_names,
+                               "script_args": script_args,
+                               "topics_to_agents": topics_to_agents,
+                               "topics_from_agents": topics_from_agents}
+
+
+        #Set a default configuration to ensure that self.configure is called immediately to setup
+        #the agent.
+        self.vip.config.set_default("config", self.default_config)
+        #Hook self.configure up to changes to the configuration file "config".
+        self.vip.config.subscribe(self.configure, actions=["NEW", "UPDATE"], pattern="config")
+
+    def configure(self, config_name, action, contents):
+        """
+        Called after the Agent has connected to the message bus. 
+        If a configuration exists at startup this will be 
+        called before onstart.
+        Is called every time the configuration in the store changes.
+        """
+        config = self.default_config.copy()
+        config.update(contents)
+
+        _log.debug("Configuring Agent")
+
+        try:
+            script_names = config["script_names"]
+            script_args = config["script_args"]
+            topics_to_agents = config["topics_to_agents"]
+            topics_from_agents = config["topics_from_agents"]
+
+        except ValueError as e:
+            _log.error("ERROR PROCESSING CONFIGURATION: {}".format(e))
+            return
+
+        self.script_names = script_names
+        self.script_args = script_args
+        self.topics_to_agents = topics_to_agents
+        self.topics_from_agents = topics_from_agents
+        self._create_subscriptions(self.topics_from_agents)
+
+        for script in range(len(self.script_names)):
```

Comment:
Thank you for the suggestion. I think in this specific case, using it as an index is easier, so it may be used to iterate through other arguments from the config, but if you know of a cleaner way to write this for loop, I am certainly open to it.